http-clients/src/main/java/org/triplea/http/client/web/socket/WebSocketConnection.java

```diff
@@ -74,31 +69,30 @@ void addListener(final WebSocketConnectionListener listener) {
 
   /** Does an async close of the current websocket connection. */
   void close() {
-    if (client.isOpen()) {
-      new Thread(
-              () -> {
-                if (client.isOpen()) {
-                  client.getConnection().close();
-                }
-              })
-          .start();
-    }
     closed = true;
+    client.close();
   }
 
+  /**
+   * Initiates a websocket connection. Must be called before {@link #sendMessage(String)}. This
+   * method blocks until the connection has either successfully been established, or the connection
+   * failed.
+   */
   void connect() {
     Preconditions.checkState(!client.isOpen());
     Preconditions.checkState(!closed);
-    webSocketConnector.initiateConnection();
+    Interruptibles.await(
+        () -> client.connectBlocking(DEFAULT_CONNECT_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS));
   }
 
+  /**
+   * Sends a message asynchronously.
+   *
+   * @throws IllegalStateException If the connection hasn't been opened yet. {@link #connect()}
+   *     needs to be called first.
+   */
   void sendMessage(final String message) {
     Preconditions.checkState(!closed);
-    webSocketConnector.waitUntilConnectionIsOpen();
-    Postconditions.assertState(Thread.currentThread().isInterrupted() || client.isOpen());
-    // if we aborted waiting for the connection, current thread will be interrupted, do a no-op.
-    if (!Thread.currentThread().isInterrupted()) {
-      client.send(message);
-    }
+    client.send(message);
```

Comment:
Any thoughts to checking `client.isOpen()`? I vaguely recall cases where that was an important check during system testing. It would be good to see any error behavior here be covered by unit tests so that we can verify the handling for these edge-cases that are hard to reproduce manually.